Include/object.h

```diff
@@ -121,6 +121,7 @@ typedef struct {
 
 #define Py_REFCNT(ob)           (_PyObject_CAST(ob)->ob_refcnt)
 #define Py_TYPE(ob)             (_PyObject_CAST(ob)->ob_type)
+#define Py_IS_TYPE(ob, tp)      (Py_TYPE(ob) == (tp))
```

Comment:
I would prefer to define it as a static inline function which would accept any type but use _PyObject_CAST() to cast the first parameter to `PyObject*`. It still needs a macro: see a few lines below.